tests/legacy-cli/e2e/tests/build/lazy-load-syntax.ts

```diff
@@ -69,22 +90,42 @@ export default async function () {
     buildTarget['configurations']['production'] = { aot: true };
   });
 
-  // Test `import()` style lazy load.
-  await replaceLoadChildren(`() => import('./lazy/lazy.module').then(m => m.LazyModule)`);
-  await ng('e2e');
-  await ng('e2e', '--prod');
-
   // Test string import with factory shims.
   await replaceLoadChildren(`'./lazy/lazy.module#LazyModule'`);
   await replaceInFile('tsconfig.app.json', `"allowEmptyCodegenFiles": false`,
     `"allowEmptyCodegenFiles": true`);
-  await expectToFail(() => ng('e2e')); // Currently broken.
-  await ng('e2e', '--prod');
+  if (ivyProject) {
+    // Ivy should not support the string syntax.
+    await expectToFail(() => ng('e2e'));
+    await expectToFail(() => ng('e2e', '--prod'));
+  } else {
+    // View engine should support the string syntax.
+    await ng('e2e');
+    await ng('e2e', '--prod');
+  }
 
   // Test string import without factory shims.
   await replaceLoadChildren(`'./lazy/lazy.module#LazyModule'`);
   await replaceInFile('tsconfig.app.json', `"allowEmptyCodegenFiles": true`,
-  `"allowEmptyCodegenFiles": false`);
-  await expectToFail(() => ng('e2e')); // Not supported.
-  await expectToFail(() => ng('e2e', '--prod')); // Not supported.
+    `"allowEmptyCodegenFiles": false`);
+  if (ivyProject) {
+    // Ivy should not support the string syntax.
+    await expectToFail(() => ng('e2e'));
+    await expectToFail(() => ng('e2e', '--prod'));
+  } else {
+    // View engine should support the string syntax.
+    await ng('e2e');
+    await ng('e2e', '--prod');
+  }
+
+  // Test `import()` style lazy load.
+  await updateJsonFile('angular.json', json => {
+    // Add the experimental flag to import factories in View Engine.
+    const buildTarget = json['projects'][projectName]['architect']['build'];
+    buildTarget['options']['experimentalImportFactories'] = true;
+  });
+  // Both Ivy and View Engine should support it.
+  await replaceLoadChildren(`() => import('./lazy/lazy.module').then(m => m.LazyModule)`);
+  await ng('e2e');
+  await ng('e2e', '--prod');
```

Comment:
This test will fail because this PR isn't using a Angular release that contains https://github.com/angular/angular/pull/29392. Once that is released this PR should be updated to use it.